### PR TITLE
[tr] Deep Learning tips and tricks

### PR DESCRIPTION
**Pull request by ayyucekizrak**
I've completed the entire translation of this title. 
Please note that the review process can be started. Thank you!

### CODE REVIEW

**Comment by shervinea** [comment]
Thanks for your hard work @ayyucekizrak!

@yavuzKomecoglu thanks for pointing the redundancy out, will fix it in later iterations.

File: tr/deep-learning-tips-and-tricks.md [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+**20. Definitions**
+
+&#10230; Tanımlamalar
+
+<br>
+
+
+**21. Epoch ― In the context of training a model, epoch is a term used to refer to one iteration where the model sees the whole training set to update its weights.**
+
+&#10230; Epoch ― Bir modelin eğitimi kapsamında, modelin ağırlıklarını güncellemek için tüm eğitim setini kullandığı bir yinelemeye ifade etmek için kullanılan bir terimdir.
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: Dönem (Epok) ― Bir modelin eğitimi kapsamında, modelin ağırlıklarını güncellemek için tüm eğitim setini kullandığı bir yinelemeye ifade etmek için kullanılan bir terimdir.

File: tr/deep-learning-tips-and-tricks.md [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+
+**28. Updating weights ― In a neural network, weights are updated as follows:**
+
+&#10230; Ağırlıkların güncellenmesi ― Bir sinir ağında, ağırlıklar aşağıdaki gibi güncellenir:
+
+<br>
+
+
+**29. [Step 1: Bir küme eğitim verisi alın ve kaybı hesaplamak için ileriye doğru ilerleyin, Step 2: Her ağırlığa göre kaybın derecesini elde etmek için kaybı tekrar geriye doğru yayın, Step 3: Use the gradients to update the weights of the network.]**
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: **29. [Step 1: Take a batch of training data and perform forward propagation to compute the loss, Step 2: Backpropagate the loss to get the gradient of the loss with respect to each weight, Step 3: Use the gradients to update the weights of the network.]**

File: tr/deep-learning-tips-and-tricks.md [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+&#10230; [Adım 1: Take a batch of training data and perform forward propagation to compute the loss, Step 2: Backpropagate the loss to get the gradient of the loss with respect to each weight, Step 3: Ağın ağırlıklarını güncellemek için gradyanları kullanın.]
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: [Adım 1: Bir küme eğitim verisi alın ve kaybı hesaplamak için ileriye doğru ilerleyin, Adım 2: Her ağırlığa göre kaybın derecesini elde etmek için kaybı tekrar geriye doğru yayın, Adım 3: Ağın ağırlıklarını güncellemek için gradyanları kullanın.]

File: tr/deep-learning-tips-and-tricks.md [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+<br>
+
+
+**30. [Forward propagation, Backpropagation, Weights update]**
+
+&#10230; [İleri yayılım, Geriye yayılım, Ağırlıkların gğncellenmesi]
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: [İleri yayılım, Geriye yayılım, Ağırlıkların güncellenmesi]

File: tr/deep-learning-tips-and-tricks.md [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+**32. Weights initialization**
+
+&#10230; Ağırlıkların başlangıçlandırılması
+
+<br>
+
+
+**33. Xavier initialization ― Instead of initializing the weights in a purely random manner, Xavier initialization enables to have initial weights that take into account characteristics that are unique to the architecture.**
+
+&#10230; Xavier başlangıcı ―  Ağırlıkları tamamen rastgele bir şekilde başlatmak yerine, Xavier ilklendirme, mimariye özgü özellikleri dikkate alan ilk ağırlıkların alınmasını sağlar.
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: Xavier başlangıcı (ilklendirme) ―  Ağırlıkları tamamen rastgele bir şekilde başlatmak yerine, Xavier başlangıcı, mimariye özgü özellikleri dikkate alan ilk ağırlıkların alınmasını sağlar.

File: tr/deep-learning-tips-and-tricks.md [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+**38. Optimizing convergence**
+
+&#10230; Yakınsamayı optimize etmek
+
+<br>
+
+
+**39. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. It can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.
+**
+&#10230; Genellikle α veya bazen η olarak belirtilen öğrenme oranı, ağırlıkların hangi hızda güncellendiğini belirler. Sabitlenebilir veya uyarlanabilir şekilde değiştirilebilir. Mevcut en popüler yöntemin adı Adam'dır ve öğrenme hızını ayarlayan bir yöntemdir.
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: Öğrenme oranı (adımı) ― Genellikle α veya bazen η olarak belirtilen öğrenme oranı, ağırlıkların hangi hızda güncellendiğini belirler. Sabitlenebilir veya uyarlanabilir şekilde değiştirilebilir. Mevcut en popüler yöntemin adı Adam'dır ve öğrenme hızını ayarlayan bir yöntemdir.

File: tr/deep-learning-tips-and-tricks.md [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+
+<br>
+
+**40. Adaptive learning rates ― Letting the learning rate vary when training a model can reduce the training time and improve the numerical optimal solution. While Adam optimizer is the most commonly used technique, others can also be useful. They are summed up in the table below:**
+
+&#10230; Uyarlanabilir öğrenmeoranları ― Bir modelin eğitilmesi sırasında öğrenme oranının değişmesine izin vermek eğitim süresini kısaltabilir ve sayısal optimum çözümü iyileştirebilir. Adam optimizasyonu yöntemi en çok kullanılan teknik olmasına rağmen, diğer yöntemler de faydalı olabilir. Bunlar aşağıdaki tabloda özetlenmiştir:
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: Uyarlanabilir öğrenme oranları ― Bir modelin eğitilmesi sırasında öğrenme oranının değişmesine izin vermek eğitim süresini kısaltabilir ve sayısal optimum çözümü iyileştirebilir. Adam optimizasyonu yöntemi en çok kullanılan teknik olmasına rağmen, diğer yöntemler de faydalı olabilir. Bunlar aşağıdaki tabloda özetlenmiştir:

File: tr/deep-learning-tips-and-tricks.md [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+
+**50. [LASSO, Ridge, Elastic Net]**
+
+&#10230; [LASSO, Ridge, Elastic Net]
+
+<br>
+
+**50 bis. Shrinks coefficients to 0, Good for variable selection, Makes coefficients smaller, Tradeoff between variable selection and small coefficients]**
+
+&#10230; bis. Katsayıları 0'a düşürür, Değişken seçimi için iyi, Katsayıları daha küçük yapar, Değişken seçimi ile küçük katsayılar arasında ödünleşim sağlar]
```
Comment (yavuzKomecoglu) [inline tr/deep-learning-tips-and-tricks.md]: [Katsayıları 0'a düşürür, Değişken seçimi için iyi, Katsayıları daha küçük yapar, Değişken seçimi ile küçük katsayılar arasında ödünleşim sağlar]

**Comment by ayyucekizrak** [comment]
@shervinea editing of all suggestions of the reviewer is completed. The file can be merged if it is ok for you. Thank you again @yavuzKomecoglu 👍

**Comment by shervinea** [comment]
Awesome! Thank you @ayyucekizrak and @yavuzKomecoglu!
